### ISSUE TITLE
Add dictation search support

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#170554]
Many modern browsers/applications offer the ability to search via dictation. It would be nice if we supported this in VS Code's command palette and potentially other search input areas.

![A screenshot of google's search bar containing a microphone toggle for searching via dictation](https://user-images.githubusercontent.com/29464607/210569310-0c289b92-010d-4a1e-9489-7fb15a9ae780.png)

**Comment by meganrogge** [comment]
@TylerLeonhardt any thoughts?

**Comment by adiabatic** [comment]
What sort of people is this feature for, and could they be better served by directing engineer time elsewhere?

Microphone buttons on search boxes make at least some sense because there are all sorts of things one might want to search for yet be unable to spell.

With a program like Visual Studio Code, though, why would someone who needs voice recognition use VS Code's own feature instead of the operating system's voice-control system, or a systemwide add-on like Dragon?

**Comment by meganrogge** [comment]
This request would benefit those who do not have or want to use an operating system's voice control system. 

It would supplement the experience of those who do use those

**Comment by mrob95** [comment]
Hi all, 

I program in VS code full-time using only dictation software (no keyboard). There's a small but dedicated community of voice-coders, mostly developers and students with some form of disability (often RSI, in my case a spinal cord injury).

I don't think this feature would be worth the effort required to implement it and I personally wouldn't use it. The reason is that, as @adiabatic identified, people who are serious about using voice rec in place of a mouse and keyboard have solutions that are systemwide and can issue input to all apps. This makes a lot more sense than having every app provide its own dictation support - even if every app did it, trying to keep track of the quirks and limitations of all the different implementations would be a nightmare.

It's awesome to see dictation support noted as a potential accessibility issue though. There is an active community of people working on improving the VSCode voice coding experience with extensions like https://github.com/cursorless-dev/cursorless. As far as I know the biggest pain point that can't be fixed through extensions is still https://github.com/microsoft/vscode/issues/97790 / https://github.com/microsoft/vscode/issues/117980. Any movement on that would be very much appreciated.

-Mike

cc @pokey

**Comment by meganrogge** [comment]
Thanks for all of the input. I'll close this issue in favor of the others cited as being more useful
